src/pypi_changes/_print.py

```diff
@@ -32,23 +33,55 @@ def print_tree(distributions: Iterable[Package], options: Options) -> None:
         text.stylize(f"link https://pypi.org/project/{pkg.name}/#history")
         text.append(" ", "white")
         text.append(pkg.version, "blue")
-        last_release = pkg.last_release
-        if last_release is not None:  # pragma: no branch
-            if last_release["version"] != pkg.version and pkg.info is not None:
-                for a_version, releases in pkg.info["releases"].items():  # pragma: no branch
-                    first_release_at = releases[0]["upload_time_iso_8601"]
-                    if a_version == pkg.dist.version and first_release_at is not None:
-                        text.append(" ")
-                        text.append(naturaltime(now - first_release_at), "green")
-                        break
-                text.append(f" remote {last_release['version']}", "red")
-            if last_release["upload_time_iso_8601"] is not None:  # pragma: no branch
```

Comment:
I'd actually like to take this refactor further and move all non-presentational logic elsewhere - a function that returns the processed dataset so that it can be fed to output formatters (rich text, json) but also consumed by 3rd party tools that use `pypi-changes` as a library.

The main point of my contribution is to improve composability of `pypi-changes` so that it can be more easily used in broader scoped tools.

Reply:
I'm open to this.

Reply:
I will not do this as part of this PR to expedite its acceptance.